Case expression support. (#2319)
confluentinc/ksql: docs/developer-guide/syntax-reference.rst

      WINDOW TUMBLING (SIZE 20 SECONDS)
      GROUP BY page_id;

CASE
~~~~

 **Synopsis**

 .. code:: sql

     CASE
        WHEN condition THEN result

>         [ WHEN ... ]

Reviewer: Can we change this to `[ WHEN ... THEN ... ]` to make it more clear that each `WHEN` must be paired with a `THEN`?

Author: Sure, added `THEN` too.

>         [ WHEN ... ]

Reviewer: Of lower priority, but it’d be nice if we could also make it more clear that any number of `WHEN … THEN …` clauses are allowed. Perhaps either
```
WHEN condition THEN result
[ WHEN … THEN … ]
…
[ WHEN … THEN … ]
[ ELSE result ]
```
or updating the example to use more than two when clauses?

Author: Added your suggestion!

        [ ELSE result ]

>     END

Reviewer 2: ```suggestion
     END
```

>  Currently, KSQL supports ``searched`` form of CASE expression. In this form, CASE evaluates

Reviewer 2: ```suggestion
 Currently, KSQL supports a ``searched`` form of CASE expression. In this form, CASE evaluates
```

> each boolean ``conditition`` in WHEN caluses from left to right. If a condition is true then it returns the

Reviewer 2: ```suggestion
each boolean ``condition`` in WHEN clauses, from left to right. If a condition is true, it returns the
```

> corresponding result. If none of the conditions are true, it returns the result from the ELSE clause.

Reviewer: Specify what happens when none of the when conditions evaluate to true, and there is no ELSE clause.

 The schema for all results should be the same, otherwise, KSQL will reject the statement.
 Here is an example of CASE expression:

 .. code:: sql

     SELECT
      CASE
        WHEN orderunits < 2.0 THEN 'small'
        WHEN orderunits < 4.0 THEN 'medium'
        ELSE 'large'
      END AS case_resault

>     FROM orders;

Reviewer 2: ```suggestion
     FROM orders;
```
